modules/updating-eus-to-eus-upgrade-cli.adoc

```diff
@@ -0,0 +1,150 @@
+// Module included in the following assemblies:
+//
+// * updating/preparing-eus-eus-upgrade.adoc
+
+:_content-type: PROCEDURE
+[id="updating-eus-to-eus-upgrade-cli_{context}"]
+= EUS-to-EUS update using the CLI
+
+.Prerequisites
+
+* Update OpenShift CLI `oc` to the target version before each update.
+
+[IMPORTANT]
+====
+It is highly discouraged to skip this prerequisite. If the OpenShift CLI `oc` is not updated to the target version before your update, unexpected issues may occur.
+====
+
+.Procedure
+
+. Update any OLM Operators to versions that are compatible with both versions you are updating to.
+
+. Verify that all machine config pools display a status of `UPDATED` and that no machine config pool displays a status of `UPDATING`.
+To view the status of all machine config pools, run the following command:
++
+[source,terminal]
+----
+$ oc get mcp
+----
++
+.Example output
++
+Output is trimmed for clarity:
++
+[source,terminal]
+----
+NAME     CONFIG                                         	UPDATED   UPDATING
+master   rendered-master-ecbb9582781c1091e1c9f19d50cf836c       True  	  False
+worker   rendered-worker-00a3f0c68ae94e747193156b491553d5       True  	  False
+----
+
+. Change to the `eus-4.12` channel. Run the following command:
++
+[source,terminal]
+----
+$ oc adm upgrade channel eus-4.12
+----
++
+[NOTE]
+====
+
+If you receive an error message indicating that `eus-4.12` is not one of the
+available channels, this indicates that Red Hat is still rolling out 4.10 to 4.12 EUS updates.
+This rollout process generally takes 45-90 days starting at the GA date.
+====
++
+
+. Pause the worker machine pools that you want to skip reboots on. Run the following commands:
++
+[NOTE]
+====
+You cannot pause the master pool.
+====
++
+[source,terminal]
+----
+$ oc patch mcp/worker --type merge --patch '{"spec":{"paused":true}}'
+----
+
+. Update to version 4.11. Run the following command:
++
+[source,terminal]
+----
+$ oc adm upgrade --to=4.11
```

Comment:
I don't think that this will work, you've got to specify a full version. I can confirm in a few minutes as I have a 4.10 cluster coming up now.

Reply:
I think we should find a way back toward `oc adm upgrade --to-latest` with the minimum z-stream of 4.10.16 being a pre-requisite, if not we suggest that they change channels to eus-4.12, apply a normal z-stream update and then continue the process.

We do not want to push everyone through 4.11.1 specifically as we've fixed hundreds of bugs since.

Reply:
> We do not want to push everyone through 4.11.1

good idea! especially since it is not guaranteed that whatever version customer cluster is at currently, will have an upgrade edge to whatever specific version we use in the doc. the only possible caveat, is that `--to-latest` does not guarantee a minor (e.g. 4.10.x to 4.11.y) upgrade. however even in this case, the reason would be lack of direct upgrade edge to next minor for current cluster version in eus-?? channel, in which case `--to` probably won't render a better outcome anyway. 
a side note, perhaps it would be interesting scenario to evaluate coverage for: an eus upgrade which jumps 2 minors, but in more than 2 upgrades with paused workers

Reply:
makes sense :) so - how does this affect the following steps? And how should we update the example outputs? I'm fine with them staying as examples, but step 5 and then step 8 both involve updating the versions

Reply:
@wking clarified this for me :) we'll need this step for both 5 and 8. All good!

Reply:
Keeping example outputs the same